python/dolfinx/wrappers/fem.cpp

```diff
@@ -564,9 +554,9 @@ void declare_objects(py::module& m, const std::string& type)
           .def_property_readonly("X",
                                  [](const dolfinx::fem::Expression<T>& self)
                                  {
-                                   return py::array_t<double>(self.X().shape(),
-                                                              self.X().data(),
-                                                              py::cast(self));
+                                   auto [X, shape] = self.X();
```

Comment:
if we use `property_readonly`, should we then move X, or should this just be a property?

Reply:
Make into a function now since it returns a new copy.